eth/src/main/kotlin/com/d3/eth/sidechain/EthChainHandler.kt

```diff
@@ -162,6 +170,54 @@ class EthChainHandler(
         }
     }
 
+    /**
+     * Tries to parse transaction as a master contract method call
+     * @param transaction Ethereum block transaction
+     * @return list of notary events on Ethereum withdrawal finalization
+     */
+    private fun handleWithdrawal(
+        transaction: Transaction,
+        time: BigInteger
+    ): List<SideChainEvent.PrimaryBlockChainEvent> {
+        logger.info { "Handle Ethereum master contract call tx ${transaction.hash}" }
+
+        val receipt = web3.ethGetTransactionReceipt(transaction.hash).send()
+
+        if (!receipt.transactionReceipt.get().isStatusOK) {
```

Comment:
Can we be sure transaction receipt is not empty here?

Reply:
Yes, since we've got this transaction from block